provider/ec2/environ_test.go

```diff
@@ -159,11 +146,20 @@ func (*Suite) TestPortsToIPPerms(c *gc.C) {
 			ToPort:    120,
 			SourceIPs: []string{"0.0.0.0/0"},
 		}},
+	}, {
+		about: "source ranges",
+		ports: []network.IngressRule{network.NewIngressRule("tcp", 80, 82, "192.168.1.0/24", "0.0.0.0/0")},
```

Comment:
The docs for various providers say it's ok to pass in source ranges which overlap - the least restrictive one is what will be used in practice. Excluding "0.0.0.0/0" means there's slightly different semantics to constructing the rules in each case. I'd like to leave just for now and add a check for 0.0.0.0/0 if we decide it's warranted when the firewaller is updated.